alvr/client_openxr/src/extra_extensions/facial_tracking_htc.rs

```diff
@@ -18,6 +21,29 @@ impl FacialTrackerHTC {
             .htc_facial_tracking
             .ok_or(sys::Result::ERROR_EXTENSION_NOT_PRESENT)?;
 
+        let props = super::get_props(
+            &session,
+            system,
+            sys::SystemFacialTrackingPropertiesHTC {
+                ty: sys::SystemFacialTrackingPropertiesHTC::TYPE,
+                next: ptr::null_mut(),
+                support_eye_facial_tracking: sys::FALSE,
+                support_lip_facial_tracking: sys::FALSE,
+            },
+        )?;
+
+        let expression_count = if facial_tracking_type == sys::FacialTrackingTypeHTC::EYE_DEFAULT
+            && props.support_eye_facial_tracking.into()
+        {
+            sys::FACIAL_EXPRESSION_EYE_COUNT_HTC
+        } else if facial_tracking_type == sys::FacialTrackingTypeHTC::LIP_DEFAULT
+            && props.support_lip_facial_tracking.into()
+        {
+            sys::FACIAL_EXPRESSION_LIP_COUNT_HTC
+        } else {
+            return Err(sys::Result::ERROR_FEATURE_UNSUPPORTED);
+        };
+
```

Comment:
This is a bit hard to understand, maybe a match would be better? And/or actually nesting the conditions

Reply:
I'm not sure you can make this any better with a match, we are checking different variables for each branch